### PR TITLE
[v1] Ref forwarding for scroll components (Fixes infinity scroll)

### LINKED ISSUE #616: FlatList onEndReached not working if onScroll is set in renderScrollComponent
```jsx
  <FlatList
    onEndReached={() => {
      console.log("reached");
    }}
    renderScrollComponent={props => {
      return (
        <Animated.ScrollView
          {...props}
          onScroll={Animated.event(
            [{ nativeEvent: { contentOffset: { y: value } } }],
            { useNativeDriver: true }
          )}
        />
      );
    }}
  />;
```

I did get the value update. However, `onEndReached` is never called. 

Any idea how to get the value update and not breaking `onEndReached`?

### PR DESCRIPTION
**Pull request by CallMeHeisenberg**
## Description

The infinity scroll was not working as expected to when creating a new FlatList with createAnimatedComponent. Once you reach the bottom and try to add more images you are automatically pushed back to the top. Similar behaviour also occurs on ScrollView.  (Using reanimated's Animated.ScrollView)

Fixes #616 .

## Changes

- Added components folder to store all the components in a clean way
- Added FlatList & SectionList components
- Moved existing animated components to the ./components folder so it matches the modified architecture

### CODE REVIEW

File: src/components/AnimatedFlatList.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import { FlatList } from 'react-native';
+import createAnimatedComponent from '../createAnimatedComponent';
+
+const FlatListWithEventThrottle = React.forwardRef((props, ref) => (
+    <FlatList scrollEventThrottle={0.0001} {...props} ref={ref} />
```
Comment (jkadamczyk) [inline src/components/AnimatedFlatList.js:6]: Why do you set scrollEventThrottle to such value? Why exactly that value?

File: src/components/AnimatedFlatList.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+));
+
+module.exports = createAnimatedComponent(
+    FlatListWithEventThrottle,
+);
```
Comment (jkadamczyk) [inline src/components/AnimatedFlatList.js:11]: Missing line break at the end of file

File: src/components/AnimatedScrollView.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import { ScrollView } from 'react-native';
+import createAnimatedComponent from '../createAnimatedComponent';
+
+const ScrollViewWithEventThrottle = React.forwardRef((props, ref) => (
+    <ScrollView scrollEventThrottle={0.0001} {...props} ref={ref} />
```
Comment (jkadamczyk) [inline src/components/AnimatedScrollView.js:6]: Why do you set scrollEventThrottle to such value? Why exactly that value?

File: src/components/AnimatedScrollView.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+));
+
+module.exports = createAnimatedComponent(
+    ScrollViewWithEventThrottle,
+);
```
Comment (jkadamczyk) [inline src/components/AnimatedScrollView.js:11]: Missing line break at the end of file

File: src/components/AnimatedSectionList.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import { SectionList } from 'react-native';
+import createAnimatedComponent from '../createAnimatedComponent';
+
+const SectionListWithEventThrottle = React.forwardRef((props, ref) => (
+    <SectionList scrollEventThrottle={0.0001} {...props} ref={ref} />
```
Comment (jkadamczyk) [inline src/components/AnimatedSectionList.js:6]: Why do you set scrollEventThrottle to such value? Why exactly that value?

File: src/components/AnimatedSectionList.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+));
+
+module.exports = createAnimatedComponent(
+    SectionListWithEventThrottle,
+);
```
Comment (jkadamczyk) [inline src/components/AnimatedSectionList.js:11]: Missing line break at the end of file

File: src/components/AnimatedFlatList.js (hunk 1 above)
Comment (jkadamczyk) [reply src/components/AnimatedFlatList.js:6]: I can now see your comment under an issue. I am hesitant about your proposal to set `scrollEventThrottle` to `0.0001` value. As stated above I am not sure why such a value.
CC @jakub-gonet

**Comment by CallMeHeisenberg** [comment]
I'll commit during this weekend the fixes ))

File: src/components/AnimatedSectionList.js (hunk 5 above)
Comment (CallMeHeisenberg) [reply src/components/AnimatedSectionList.js:6]: It is exactly the same as react-native's SectionList Animated component and I remember seeing some bug without it, just can't pop into my mind which it was ))

**Comment by CoryWritesCode** [comment]
@CallMeHeisenberg any movement on this?

File: src/components/AnimatedFlatList.js (hunk 1 above)
Comment (jakub-gonet) [reply src/components/AnimatedFlatList.js:6]: For me, it looks like a hack covering the underlying issue.

**Comment by jakub-gonet** [comment]
Seems like this PR has been abandoned. If @CallMeHeisenberg want to continue working on this please respond, I'll reopen.

**Pull request closed without merging**